src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -390,16 +390,16 @@ ParsedQuery Visitor::visit(Parser::DescribeQueryContext* ctx) {
   // Convert the describe resources (variables or IRIs) from the format that the
   // parser delivers to the one that the `parsedQuery::Describe` struct expects.
   std::vector<Variable> describedVariables;
-  for (GraphTerm& resource : describedResources) {
+  for (VarOrIri& resource : describedResources) {
     if (std::holds_alternative<Variable>(resource)) {
       const auto& variable = std::get<Variable>(resource);
       describeClause.resources_.emplace_back(variable);
       describedVariables.push_back(variable);
     } else {
-      AD_CORRECTNESS_CHECK(std::holds_alternative<Iri>(resource));
-      auto iri =
-          TripleComponent::Iri::fromIriref(std::get<Iri>(resource).toSparql());
-      describeClause.resources_.emplace_back(std::move(iri));
+      AD_CORRECTNESS_CHECK(
+          std::holds_alternative<TripleComponent::Iri>(resource));
+      describeClause.resources_.emplace_back(
+          std::get<TripleComponent::Iri>(resource));
```

Comment:
If it really now is only `VarOrIri` as a variant, 
then this can become `std::visit` which exhaustively checks the handled  cases at compile time.
Thank you for cleaning up QLever along the way.